packages/flutter_tools/templates/skeleton/lib/src/app.dart.tmpl

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import 'dummy_feature/dummy_item_details_view.dart';
+import 'dummy_feature/dummy_item_list_view.dart';
+import 'settings/settings_controller.dart';
+import 'settings/settings_view.dart';
+
+/// The Widget that configures your application.
+class MyApp extends StatelessWidget {
+  const MyApp({
+    Key? key,
+    required this.settingsController,
+  }) : super(key: key);
+
+  final SettingsController settingsController;
+
+  @override
+  Widget build(BuildContext context) {
+    // Glue the SettingsController to the MaterialApp.
+    //
+    // The AnimatedBuilder Widget listens to the SettingsController for changes.
+    // Whenever the user updates their settings, the MaterialApp is rebuilt.
+    return AnimatedBuilder(
+      animation: settingsController,
+      builder: (BuildContext context, Widget? child) {
+        return MaterialApp(
+          // Providing a restorationScopeId allows the Navigator built by the
+          // MaterialApp to restore the navigation stack when a user leaves and
+          // returns to the app after it has been killed while running in the
+          // background.
+          restorationScopeId: 'app',
+
+          // Provide the generated AppLocalizations to the MaterialApp. This
+          // allows descendant Widgets to display the correct translations
+          // depending on the user's locale.
+          localizationsDelegates: const [AppLocalizations.delegate],
```

Comment:
I think we also should also add `GlobalMaterialLocalizations.delegate` and `GlobalWidgetsLocalizations.delegate` here to avoid tripping users up if they start adding another language. (maybe even `GlobalCupertinoLocalizations.delegate` as well)

/cc @HansMuller

Reply:
Also: Should we explicitly specify the `supportedLocales` parameter here (and that it to its default `en` locale) to make users aware that if they want to support more languages, they'd have to change that list?

Reply:
Yes. Best to follow the guidelines in https://flutter.dev/docs/development/accessibility-and-localization/internationalization and point developers at that doc for more info.

Reply:
Ack. I'll add these.